api_server.py

```
@@ -74,38 +87,85 @@ def format_hours(minutes: Optional[int]) -> Optional[str]:
-            display += " today"
-        elif end_time.date() == (now + timedelta(days=1)).date():
-            display += " tomorrow"
-        else:
-            display += end_time.strftime(" on %d/%m")
+    if conn is None:
+        with get_db() as db_conn:
+            return _get_availability_with_conn(entity_id, table, now, db_conn, col)
+    return _get_availability_with_conn(entity_id, table, now, conn, col)
+
+
+def _get_availability_with_conn(entity_id, table, now, conn, col):
```

Ruff (0.15.1)

[warning] Missing return type annotation for private function `_get_availability_with_conn`

(ANN202)